test/ui/FacetValueSuggestionsProviderTest.ts

```diff
@@ -0,0 +1,128 @@
+import { FacetValueSuggestionsProvider } from '../../src/ui/FacetValueSuggestions/FacetValueSuggestionsProvider';
+import * as Mock from '../MockEnvironment';
+import { IFieldOption } from '../../src/ui/Base/ComponentOptions';
+import { QueryController, QueryStateModel, SearchEndpoint } from '../Test';
+import { IIndexFieldValue } from '../../src/rest/FieldValue';
+import { IListFieldValuesRequest } from '../../src/rest/ListFieldValuesRequest';
+
+export function FacetValueSuggestionsProviderTest() {
+  describe('FacetValueSuggestionsProvider', () => {
+    let test: FacetValueSuggestionsProvider;
+    let queryController: QueryController = Mock.mock(QueryController);
+    let queryStateModel: QueryStateModel = Mock.mock(QueryStateModel);
+    let searchEndpointMock: SearchEndpoint = Mock.mock(SearchEndpoint);
+    const someField: IFieldOption = '@bloupbloup';
+    const valueToSearch = 'cowboy';
+    const referenceFieldNumberOfResults = 10;
+    const suggestion = 'suggestion';
+
+    const setUpLastQuery = (aq: string = '', cq: string = '') => {
+      (<jasmine.Spy>queryController.getLastQuery).and.returnValue({
+        aq,
+        cq
+      });
+    };
+
+    const setUpFieldValuesBatchResponse = (values: IIndexFieldValue[][]) => {
+      const valuesWithReference: IIndexFieldValue[][] = [].concat(values).concat([getReferenceBatchResponse()]);
+      (<jasmine.Spy>searchEndpointMock.listFieldValuesBatch).and.returnValue(Promise.resolve(valuesWithReference));
+    };
+
+    const getIndexFieldValue = (numberOfResults: number, value: string) => {
+      return <IIndexFieldValue>{
+        numberOfResults,
+        value: value
+      };
+    };
+
+    const getReferenceBatchResponse = () => {
+      return <IIndexFieldValue[]>[getIndexFieldValue(referenceFieldNumberOfResults, suggestion)];
+    };
+
+    beforeEach(() => {
+      queryController = Mock.mock(QueryController);
```

Comment:
Could we use a `new MockEnvironmentBuilder().build()`  for all of those ? And then add the needed spies.